### PR TITLE
test: Upload payload size to angular firebase database

### PR DESCRIPTION
**Pull request by tinayuangao**
- Secure the firebase access token
- Upload compressed/uncompressed payload size to angular-payload-size firebase database
- Path is /payload/cli/$branch/$commit
- With commit message, and timestamp

### CODE REVIEW

File: tests/e2e/tests/build/bundle-size.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  await expectGlobFileSizeToBeUnder('dist/main.*.js', 152200 * 1.01);
+  await expectGlobFileSizeToBeUnder('dist/*', 232000 * 1.01);
+
+  // Upload payload size to firebase
+  const payloadData = await uploadBundleJsFileSize('dist/*.bundle.js');
+
+  // Add timestamp and commit message
+  payloadData['timestamp'] = Math.floor(Date.now()/1000);
+  payloadData['message'] = process.env['TRAVIS_COMMIT_MESSAGE']
+      .replace(/\\/g, '\\').replace(/"/g, '\"').replace(/'/g, "\'");
```
Comment (hansl) [inline tests/e2e/tests/build/bundle-size.ts]: Those replacements sound wrong. Seems that if you want to replace `\` with `\\` you could use `JSON.stringify(process.env['TRAVIS_COMMIT_MESSAGE'])`.

File: tests/e2e/tests/build/bundle-size.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  // The database path will be /payload/cli/$branch/$commit
+  const branchName = process.env['TRAVIS_BRANCH'].replace(/\./g, '-');
+  const dbPath = `/payload/cli/${branchName}/${process.env['TRAVIS_COMMIT']}`;
+  const payloadText = JSON.stringify(payloadData);
+
+  // Log the data
+  console.log(`The data for cli branch ${branchName} commit ${process.env['TRAVIS_COMMIT']} is:`);
+  console.log(payloadData);
+  if (process.env['TRAVIS_PULL_REQUEST'] === false) {
+    exec('firebase', 'database:update', '--data', payloadText, '--project', PROJECT_ID, '--confirm',
```
Comment (hansl) [inline tests/e2e/tests/build/bundle-size.ts]: `await exec`.

**Comment by hansl** [comment]
The message should probably be either `build:` or `test:`. `misc()` isn't a valid scope.

**Comment by tinayuangao** [comment]
Comments addressed. Update the test with Appveyor environment variables. Please take a look. Thanks!

**Comment by filipesilva** [comment]
@tinayuangao can you rebase please?

**Comment by tinayuangao** [comment]
@filipesilva Rebased. Thanks!

**Comment by filipesilva** [comment]
@hansl some new problematic licenses are being introduced with this PR:
```
Invalid package licences found:
cycle@1.0.3: ["Custom: https://github.com/douglascrockford/JSON-js"]
didyoumean@1.2.1: ["Apache"]
firebase@2.4.2: ["Custom: https://www.firebase.com/docs/web/changelog.html"]
jju@1.3.0: ["WTFPL"]
log-driver@1.2.5: ["Custom: http://linuxcommand.org/man"]
revalidator@0.1.8: ["Apache"]
```

**Comment by alexeagle** [comment]
Let's do this with the size tracker built into the ngbot instead.

**Pull request closed without merging**